### PR TITLE
Add data connection to remote map sink [HZ-3479]

### PR DESCRIPTION
**Pull request by orcunc**
Jira : https://hazelcast.atlassian.net/browse/HZ-3479

This PR adds HZ Data connection to remote map sink.

Summary of Changes :

- The tests in Sinks related to remote map are moved into a new Test. New tests are added for data connection
- Two builders have been defined . 
 * Builder for local/remote map sink/update/merge operations
 * Builder for local/remote map entry processor operations
 
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Comment by frant-hartm** [comment]
run-lab-run

**Review by frant-hartm — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/jet/impl/connector/AbstractHazelcastConnectorSupplier.java [hunk 1]
```diff
@@ -67,7 +73,23 @@ protected Processor createProcessor(HazelcastInstance instance, SerializationSer
...
+    private void createHzClient(ProcessorSupplier.Context context) {
+        // The order is important.
+        // If dataConnectionConfig is specified prefer it to clientXml
+        if (dataConnectionName != null) {
+            HazelcastDataConnection hazelcastDataConnection = context
+                    .dataConnectionService()
+                    .getAndRetainDataConnection(dataConnectionName, HazelcastDataConnection.class);
+            try {
+                instance =  hazelcastDataConnection.getClient();
+                serializationService = ((HazelcastClientProxy) instance).getSerializationService();
```
Comment (TomaszGaweda) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/connector/AbstractHazelcastConnectorSupplier.java:88]: I would not rely on technical details here; is there a reason why we cannot get serializationService in other way?

File: hazelcast/src/main/java/com/hazelcast/jet/pipeline/DataConnectionRef.java [hunk 2]
```diff
@@ -19,25 +19,25 @@
...
-@Beta
 public class DataConnectionRef implements Serializable {
 
     private static final long serialVersionUID = 1L;
 
     private final String name;
 
     public DataConnectionRef(String name) {
-        this.name = name;
+        this.name = requireNonNull(name);
```
Comment (TomaszGaweda) [inline hazelcast/src/main/java/com/hazelcast/jet/pipeline/DataConnectionRef.java:40]: please add message to the invariant

File: hazelcast/src/main/java/com/hazelcast/jet/pipeline/MapSinkBuilder.java [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+     * The function must be {@link java.io.Serializable}.
+     * <p>
+     * The given functions must be stateless and {@linkplain
+     * Processor#isCooperative() cooperative}.
+     *
+     * @param updateFn function that receives the existing map value and the item
+     *                 and returns the new map value
+     */
+    public MapSinkBuilder<T, K, V> updateFn(BiFunctionEx<? super V, ? super T, ? extends V> updateFn) {
+        checkSerializable(updateFn, "updateFn");
```
Comment (TomaszGaweda) [inline hazelcast/src/main/java/com/hazelcast/jet/pipeline/MapSinkBuilder.java:178]: checkNonNullAndSerializable?

**Review by TomaszGaweda — APPROVED** [review]
Looks good overall, I have some comments, but I am fine with implementing them as follow-up

**Comment by TomaszGaweda** [comment]
run-lab-run

**Pull request merged**